### PR TITLE
Restructure Documents folders, separate design from compliance requirements, etc.

### PR DESCRIPTION
**Pull request by chrispsommers**
Created new, structured Documents hierarchy and ToC, moved various images under Documents/images/XXX into respective per-topic image folders, drew new document relationship diagram.

### CODE REVIEW

File: Documentation/README.md [hunk 1]
```diff
@@ -1,17 +1,51 @@
...
-| [SDN-Features-Packet-Transforms](sdn-features-packet-transforms.md) | Scenarios and background. Service descriptions. Packet encapsulation formats and transformations. Scaling and performance requirements |
\ No newline at end of file
+## Organization of Documents
+Documentation consists of separate, but related *System Descriptions* (HLDs, architecture, theory of operations, etc.) and *Compliance Requirements* (hard specifications, typically numerical  but also behavioral). These two types of documents are deliberately kept separated, see [Relationships and Flow of Documents](#relationships-and-flow-of-documents).
+
+Documentation is organized into folders as follows. Each feature or topic has all the high-level specs  and the compliance requirements in the same parent folder, e.g. General, High-Availability, etc., making it easier to access related information about one topic. As the complexity grows, this helps keep things organized according to "functional topic." 
+```
+topic1
+    design
+        Topic1 High-level Descriptions and architecture
```
Comment (KrisNey-MSFT) [inline Documentation/README.md:15]:
```suggestion
        topic1 High-level Descriptions and architecture
```

File: Documentation/README.md [hunk 2]
```diff
@@ -1,17 +1,51 @@
...
+    requirements
+        Topic1 compliance Requirements
```
Comment (KrisNey-MSFT) [inline Documentation/README.md:17]:
```suggestion
        topic1 compliance Requirements
```

File: Documentation/README.md [hunk 3]
```diff
@@ -1,17 +1,51 @@
...
+| General Architecture and Requirements| [Parent Folder](general/README.md) \| [ Design](general/design/README.md) \| [Compliance Requirements](general/requirements/README.md)|
+| High-Availability (HA)                                                    | [Parent Folder](high-avail/README.md) \| [ Design](high-avail/design/README.md) \| [Compliance Requirements](high-avail/requirements/README.md)|
+| Load Balancer Service                | [Parent Folder](load-bal-service/README.md) \| [ Design](load-bal-service/design/README.md) \| [Compliance Requirements](load-bal-service/requirements/README.md)|
+| VNET-to-VNET Service                         | [Parent Folder](vnet2vnet-service/README.md) \| [ Design](vnet2vnet-service/design/README.md) \| [Compliance Requirements](vnet2vnet-service/requirements/README.md)|
+| SAI Southbound API                   | [Parent Folder](sai/README.md) \| [ Design](sai/design/README.md) \| [Compliance Requirements](sai/requirements/README.md)|
+| gNMI Northbound API                  | [Parent Folder](gnmi/README.md) \| [ Design](gnmi/design/README.md) \| [Compliance Requirements](gnmi/requirements/README.md)|
+
```
Comment (KrisNey-MSFT) [inline Documentation/README.md:39]:
```suggestion
| Service Tunnel & Private Link
| VNET Peering
| Encryption Gateway
| ER Gateway
```

File: documentation/README.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+| Service Tunnel & Private Link Service                          | [Parent Folder](stpl-service/README.md) \| [ Design](stpl-service/design/README.md) \| [Compliance Requirements](stpl-service/requirements/README.md)|
+| VNET Peering Service                          | [Parent Folder](vnet-peering-service/README.md) \| [ Design](vnet-peering-service/design/README.md) \| [Compliance Requirements](vnet-peering-service/requirements/README.md)|
+| Express Route (ER) Service                    |  [Parent Folder](express-route-service/README.md) \| [ Design](express-route-service/design/README.md) \| [Compliance Requirements](express-route-service/requirements/README.md)|
+| Encryption Gateway Service                    |  [Parent Folder](encrypt-gw-service/README.md) \| [ Design](encrypt-gw-service/design/README.md) \| [Compliance Requirements](encrypt-gw-service/requirements/README.md)|
+
+# Relationships and Flow of Documents
+The diagram below shows how High-Level Descriptions beget Compliance requirements, compliance requirements beget test cases, and test cases are executed by test scripts to produce Test Results.
+
+![dash-specs-flow](images/general/dash-specs-flow.svg).
+
```
Comment (mmiele) [inline documentation/README.md:59]: I am wondering if we can make this a clickable image, where each link point to the related article(s).

Comment (chrispsommers) [reply documentation/README.md:59]: This is just a general flow, there are no specific articles per se to click on. Eventually there will be many artifacts at each "level." There could be hundreds of test cases for example.

File: documentation/README.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+Some of the guiding principles for this aproach are:
```
Comment (mmiele) [inline documentation/README.md:60]:
```suggestion
The following are some of the guiding principles for this approach:
```

Comment (chrispsommers) [reply documentation/README.md:60]: I don't agree with your proposed change, I wrote as if one is reading the final document, not to describe "what I am changing in the old document." That only has historical passing value.

File: documentation/README.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+* Define the objectives and design or proposal without clutter from performance and requirement details.
```
Comment (mmiele) [inline documentation/README.md:61]:
```suggestion
* Define the objectives and the design or proposal separately from performance and requirement details.
```

File: documentation/README.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+* Express hard requirements separately from the design and architecture descriptions so that the requirements are easy to define, maintain and reference from other downstream "consumers," e.g. **test cases**. All requirements are identified with some designator which allows traceability in test cases, scripts and results.
```
Comment (mmiele) [inline documentation/README.md:62]:
```suggestion
* Describe **hard requirements** separately from the design and architecture descriptions This allows the requirements to be easily defined, maintained, and referenced from other downstream "consumers," e.g., **test cases**. All requirements must be identified with some designator which allows traceability in test cases, scripts and results.
```

File: documentation/README.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+* Strive for simultaneous human- and machine-readable data which can drive test cases. Avoid burying test parameters into the test script so that requirements can be maintained/defined independently from the (often complex) code which executes tests. Many projects exist where only a programmer can locate and ferret out actual test criteria, often expressed as hard-coded constants buried within thousands of lines of test automation code. In contrast, these criteria should be easily accessible, reviewable and maintainable, to anyone familiar with the project.
```
Comment (mmiele) [inline documentation/README.md:63]:
```suggestion
* We encourage the creation of simultaneous **human** and **machine-readable** data which can **drive test cases**.  
* We must avoid burying test parameters into the test scripts. This allows the requirements to be maintained/defined independently from the (often complex) code which executes tests. 
* Many projects exist where only a programmer can locate and ferret out actual test criteria, often expressed as hard-coded constants buried within thousands of lines of test automation code. For quality control, these criteria must be easily accessible, reviewable and maintainable, to anyone familiar with the project.
```

File: documentation/README.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+* Complete auditability and tracebility of tests cases, test results, associated specs and DUT/SUT configuration. This means  a test run will record versions of every item including Git Repo commit SHA ids, branches, tags, SW versions, API versions, etc.
```
Comment (mmiele) [inline documentation/README.md:64]:
```suggestion
* We advocate complete auditability and traceability of tests cases, test results, associated specs and DUT/SUT configuration. This means a test run will record versions of every item including GitHub Repo commit SHA ids, branches, tags, SW versions, API versions, etc.
```

File: documentation/README.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+* Clear, concise human-readable reports, plus machine-readable results allowing dashboards, rolling-up of results, etc.
```
Comment (mmiele) [inline documentation/README.md:65]:
```suggestion
* Clear, concise and to the point human-readable reports, plus machine-readable results allowing dashboards, rolling-up of results, etc.
```

File: documentation/general/design/README.md [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+# DASH General Design Documents
+
+This folder contains DASH "general" design and architecture documents.
+
+# Contents
+
+| Document                                               | Description                                |
+| ------------------------------------------------------ | ------------------------------------------ |
+| [dash-high-level-design.md](dash-high-level-design.md) | DASH High-Level Architecture and Design |
+| [sdn-features-packet-transforms.md](sdn-features-packet-transforms.md) | DASH VNET-to-VNET HLD and Packet Transforms   |
```
Comment (mmiele) [inline documentation/general/design/README.md:14]: - [x] Links OK

File: documentation/general/README.md [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+# DASH General Documents
+
+This folder contains DASH "General" design and requirements documents.
+
+# Contents
+
+| Folder                                                 | Description                                  |
+| ------------------------------------------------------ | -------------------------------------------- |
+| [design](design/README.md)                             | DASH General design & architecture documents |
+| [requirements](requirements/README.md)                 | DASH General requirements documents         |
```
Comment (mmiele) [inline documentation/general/README.md:12]: - [x] Links OK

File: documentation/general/requirements/README.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+# DASH General Requirements Documents
+
+This folder contains DASH "general" compliance requirements documents.
+
+# Contents
+
+| Document                                               | Description                                |
+| ------------------------------------------------------ | ------------------------------------------ |
+| [dash-general-compliance-requirements.md](dash-general-compliance-requirements.md) | DASH General Compliance Requirements    |
+| [program-scale-testing-requirements-draft.md](program-scale-testing-requirements-draft.md) | DASH scale and performance requirements   |
```
Comment (mmiele) [inline documentation/general/requirements/README.md:14]: - [x] Links OK

File: documentation/dataplane/README.md [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+
+# DASH Dataplane Documents
+This folder contains DASH dataplane design and requirements documents.
+
+# Contents
+
+| Folder                                                 | Description                                  |
+| ------------------------------------------------------ | -------------------------------------------- |
+| [design](design/README.md)                             | DASH Dataplane design & architecture documents |
+| [requirements](requirements/README.md)                 | DASH Dataplane requirements documents         |
```
Comment (mmiele) [inline documentation/dataplane/README.md:11]: - [x] Links OK

File: documentation/high-avail/README.md [hunk 15]
```diff
@@ -0,0 +1,12 @@
...
+# DASH High-Availability Documents
+
+This folder contains DASH High Availability design and requirements documents.
+
+# Contents
+
+| Folder                                                 | Description                                  |
+| ------------------------------------------------------ | -------------------------------------------- |
+| [design](design/README.md)                             | DASH High Availability design & architecture documents |
+| [requirements](requirements/README.md)                 | DASH High Availability requirements documents         |
```
Comment (mmiele) [inline documentation/high-avail/README.md:12]: - [x] Links OK

File: documentation/high-avail/design/README.md [hunk 16]
```diff
@@ -0,0 +1,13 @@
...
+
+# DASH High-Availability Design Documents
+
+This folder contains DASH High Avalability and Scale design and architecture documents.
+
+# Contents
+
+| Document                                               | Description                                |
+| ------------------------------------------------------ | ------------------------------------------ |
+| [high-availability-and-scale.md](high-availability-and-scale.md) | DASH High-Availability and Scale design document   |
```
Comment (mmiele) [inline documentation/high-avail/design/README.md:13]: - [x] Links OK

File: documentation/gnmi/README.md [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+
+This folder contains DASH gNMI Interface design and requirements documents.
+
+
+# Contents
+
+| Folder                                                 | Description                                  |
+| ------------------------------------------------------ | -------------------------------------------- |
+| [design](design/README.md)                             | DASH gNMI design & architecture documents |
+| [requirements](requirements/README.md)                 | DASH gNMI requirements documents         |
```
Comment (mmiele) [inline documentation/gnmi/README.md:13]: - [x] Links OK

File: documentation/sai/README.md [hunk 18]
```diff
@@ -0,0 +1,12 @@
...
+# DASH SAI Documents
+
+This folder contains DASH SAI Interface design and requirements documents.
+
+# Contents
+
+| Folder                                                 | Description                                  |
+| ------------------------------------------------------ | -------------------------------------------- |
+| [design](design/README.md)                             | DASH SAI design & architecture documents |
+| [requirements](requirements/README.md)                 | DASH SAI requirements documents         |
```
Comment (mmiele) [inline documentation/sai/README.md:12]: - [x] Links OK

File: documentation/sai/design/README.md [hunk 19]
```diff
@@ -0,0 +1,13 @@
...
+
+# DASH SAI Design Documents
+
+This folder contains DASH SAI southbound API design and architecture documents.
+
+# Contents
+
+| Document                                               | Description                                |
+| ------------------------------------------------------ | ------------------------------------------ |
+| [DASH-apis-rough-draft.xlsx](DASH-apis-rough-draft.xlsx) | Preliminary SAI APIs (**NOTE** - obsolete?)   |
```
Comment (mmiele) [inline documentation/sai/design/README.md:13]: - [x] Links OK

File: documentation/load-bal-service/README.md [hunk 20]
```diff
@@ -0,0 +1,10 @@
+[ [ << Back to DASH top-level Documents](../README.md#contents) ]
+
+This folder contains DASH Load Balancer Service design and requirements documents.
+
+# Contents
+
+| Folder                                                 | Description                                  |
+| ------------------------------------------------------ | -------------------------------------------- |
+| [design](design/README.md)                             | DASH Load Balancer Service design & architecture documents |
+| [requirements](requirements/README.md)                 | DASH Load Balancer Service requirements documents         |
```
Comment (mmiele) [inline documentation/load-bal-service/README.md:10]: - [x] Links OK

File: documentation/load-bal-service/design/README.md [hunk 21]
```diff
@@ -0,0 +1,11 @@
...
+
+[ [ << Back to DASH top-level Documents](../../README.md#contents) ]
+
+This folder contains DASH Load-Balancer Service design and architecture documents.
+
+# Contents
+
+| Document                                               | Description                                |
+| ------------------------------------------------------ | ------------------------------------------ |
+| [load-balancer-v3.md](load-balancer-v3.md) | DASH Load Balancer design document   |
```
Comment (mmiele) [inline documentation/load-bal-service/design/README.md:11]: - [x] Links OK

File: documentation/vnet2vnet-service/README.md [hunk 22]
```diff
@@ -0,0 +1,12 @@
...
+# DASH VNET-to-VNET Service Documents
+
+This folder contains DASH VNET-to-VNET Service design and requirements documents.
+
+# Contents
+
+| Folder                                                 | Description                                  |
+| ------------------------------------------------------ | -------------------------------------------- |
+| [design](design/README.md)                             | DASH VNET-to-VNET Service design & architecture documents |
+| [requirements](requirements/README.md)                 | DASH VNET-to-VNET Service requirements documents         |
```
Comment (mmiele) [inline documentation/vnet2vnet-service/README.md:12]: - [x] Links OK

File: documentation/express-route-service/README.md [hunk 23]
```diff
@@ -0,0 +1,10 @@
+[ [ << Back to DASH top-level Documents](../README.md#contents) ]
+
+This folder contains DASH Express Route Gateway Service design and requirements documents.
+
+# Contents
+
+| Folder                                                 | Description                                  |
+| ------------------------------------------------------ | -------------------------------------------- |
+| [design](design/README.md)                             | DASH Express Route Gateway Service design & architecture documents |
+| [requirements](requirements/README.md)                 | DASH Express Route Gateway Service requirements documents         |
```
Comment (mmiele) [inline documentation/express-route-service/README.md:10]: - [x] Links OK

File: documentation/encrypt-gw-service/README.md [hunk 24]
```diff
@@ -0,0 +1,10 @@
+[ [ << Back to DASH top-level Documents](../README.md#contents) ]
+
+This folder contains DASH Encryption Gateway Service design and requirements documents.
+
+# Contents
+
+| Folder                                                 | Description                                  |
+| ------------------------------------------------------ | -------------------------------------------- |
+| [design](design/README.md)                             | DASH Encryption Gateway Service design & architecture documents |
+| [requirements](requirements/README.md)                 | DASH Encryption Gateway Service requirements documents         |
```
Comment (mmiele) [inline documentation/encrypt-gw-service/README.md:10]: - [x] Links OK

**Review by mmiele — APPROVED** [review]

**Pull request merged**